server/app/lib/actions/fusor/deployment/openstack/cfme_launch.rb

```diff
@@ -0,0 +1,108 @@
+#
+# Copyright 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+require 'fog'
+
+module Actions
+  module Fusor
+    module Deployment
+      module OpenStack
+        #Setup and Launch CFME VM
+        class CfmeLaunch < Actions::Base
+          def humanized_name
+            _('Setup and Launch CFME VM')
+          end
+
+          def plan(deployment)
+            plan_self(deployment_id: deployment.id)
+          end
+
+          def run
+            Rails.logger.debug '====== CFME Launch run method ======'
+            deployment = ::Fusor::Deployment.find(input[:deployment_id])
+            create_image(deployment)
+            create_compute_profile(deployment)
+            host = create_host(deployment)
+            deployment.cfme_address = host.ip
+            deployment.save!
+            Rails.logger.debug '====== Leaving Launc Upload run method ======'
+          end
+
+          def cfme_launch_completed
+            Rails.logger.info 'CFME Launch Completed'
+          end
+
+          def cfme_launch_failed
+            fail _('CFME Launch failed')
+          end
+
+          private
+
+          def create_image(deployment)
+            cr = ComputeResource.find_by_name("#{deployment.name}-RHOS")
+            Image.create("name" => "#{deployment.name}-cfme",
+              "username" => 'root',
+              "user_data" => 1,
+              "uuid" => cr.available_images.find { |hash| "#{deployment.name}-cfme" == hash.name }.id,
+              "compute_resource_id" => cr.id,
+              "operatingsystem_id" => Operatingsystem.find_by_title('RedHat 7.1')['id'],
```

Comment:
One of the areas of cleanup that needs to be done. 

It actually needs to exist as an operating system in Satellite with valid media, templates. and so on specified. We may actually want/need to sync 7.1 content regardless of If we install OSP so we get the operating system created in the proper way so we can launch CFME, even though we don't actually use any of it... 

But yes, after that it would make sense to tie in in any smarter way that we can, rather than hard coding the value.